### PR TITLE
Modernize rtmidi for conan v2

### PR DESCRIPTION
**Pull request by MartinDelille**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/rtmidi/all/conanfile.py [hunk 1]
```diff
@@ -75,7 +75,7 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, "share"))
 
     def package_info(self):
-        self.cpp_info.names["pkg_config"] = "rtmidi"
+        self.cpp_info.set_property("pkg_config_name", "rtmidi")
```
Comment (SpaceIm) [inline recipes/rtmidi/all/conanfile.py:78]:
```suggestion
        self.cpp_info.set_property("cmake_file_name", "RtMidi")
        self.cpp_info.set_property("cmake_target_name", "RtMidi::rtmidi")
        self.cpp_info.components["librtmidi"].set_property("cmake_target_name", "RtMidi::rtmidi")
        self.cpp_info.set_property("pkg_config_name", "rtmidi")
        self.cpp_info.components["librtmidi"].set_property("pkg_config_name", "rtmidi")
```

File: recipes/rtmidi/all/patches/4.0.0-0003-use-rtmidi-folder.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 28ee1f8..37855fe 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -231,7 +231,7 @@ install(TARGETS ${LIB_TARGETS}
+         LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+         ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+         RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+-        PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
++        PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/rtmidi)
```
Comment (SpaceIm) [inline recipes/rtmidi/all/patches/4.0.0-0003-use-rtmidi-folder.patch:10]: ok, but I think that include convention for 5.0.0 is broken: https://github.com/thestk/rtmidi/issues/285

You should set `includedirs` in `package_info()` to `os.path.join("include", "rtmidi")`

Comment (MartinDelille) [reply recipes/rtmidi/all/patches/4.0.0-0003-use-rtmidi-folder.patch]: Done here: 31a493f53629e29375ae5c471ec9b1e27ab998ac

File: recipes/rtmidi/all/conanfile.py [hunk 3]
```diff
@@ -75,7 +75,14 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, "share"))
 
     def package_info(self):
-        self.cpp_info.names["pkg_config"] = "rtmidi"
+        if(tools.Version(self.version) >= "5.0.0"):
+            self.cpp_info.components["librtmidi"].includedirs.append(os.path.join("include", "rtmidi"))
```
Comment (SpaceIm) [inline recipes/rtmidi/all/conanfile.py:79]:
```suggestion
        if tools.Version(self.version) >= "5.0.0"):
            self.cpp_info.components["librtmidi"].includedirs = [os.path.join("include", "rtmidi")]
```

File: recipes/rtmidi/all/patches/4.0.0-0003-use-rtmidi-folder.patch (hunk 2 above)
Comment (SpaceIm) [reply recipes/rtmidi/all/patches/4.0.0-0003-use-rtmidi-folder.patch]: ok, you just have to update include in test package now

**Review by SpaceIm — APPROVED** [review]

**Review by ericLemanissier — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**